QandA/QandA.Rmd

```diff
@@ -27,61 +27,243 @@ htmltools::img(src = knitr::image_uri("r4ds.png"),
 copy-on-modify vs copy-in-place: is one more preferable in certain situations?
 :::
 
-XXX
+modify in place only happens when objects with a single binding get a special performance optimisation and to enviornments.
 
 ## 2.3.2 Function calls
 
 :::question
 Can we go over and break down figure in 2.3.2
 :::
 
-XXX
+When you create this function:
+
+```{r, eval=FALSE}
+crazyfunction <- function(eh) {eh}
+```
+
+`eh` doesn't exist in memory at this point.
+
+```{r, eval=FALSE}
+x <- c(1,2,3)
+```
+
+x exists in memory.
+
+```{r, eval=FALSE}
+z <- crazyfunction(x) 
+```
+
+`z` now points at `x`, and `eh` still doesn't exist (except metaphorically in Canada). `eh` was created and exists WHILE `crazyfunction()` was being run, but doesn't get saved to the global environment, so after the function is run you can't see its memory reference. 
```

Comment:
Lol you kept this exactly 😂

Reply:
I mean I loved it - hope that's okay! Feel free to edit...